tasks/verifySpokePool.ts

```diff
@@ -0,0 +1,73 @@
+import assert from "assert";
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { getDeployedAddress } from "../src/DeploymentUtils";
+import { SpokePool__factory } from "../typechain";
+import { CHAIN_IDs, MAINNET_CHAIN_IDs, TOKEN_SYMBOLS_MAP } from "../utils/constants";
+
+const { MAINNET, POLYGON, SEPOLIA } = CHAIN_IDs;
+const TEXT_PADDING = 36;
+const CHAIN_PADDING = Object.values(MAINNET_CHAIN_IDs)
+  .sort((x, y) => x - y)
+  .pop()
+  .toString().length;
+const formatChainId = (chainId: number): string => chainId.toString().padStart(CHAIN_PADDING, " ");
+
+task("verify-spokepool", "Verify the configuration of a deployed SpokePool")
+  .addFlag("routes", "Dump deposit route history (if any)")
+  .setAction(async function (args, hre: HardhatRuntimeEnvironment) {
+    const { ethers, companionNetworks, getChainId, network } = hre;
+    const spokeChainId = parseInt(await getChainId());
+    const hubChainId = Object.values(MAINNET_CHAIN_IDs).includes(spokeChainId) ? MAINNET : SEPOLIA;
+
+    const spokeAddress = getDeployedAddress("SpokePool", spokeChainId, true);
+    const hubAddress = getDeployedAddress("HubPool", hubChainId, true);
+
+    const provider = new ethers.providers.StaticJsonRpcProvider(network.config.url);
+
+    // Initialize contracts:
+    const spokePool = await SpokePool__factory.connect(spokeAddress, provider);
+
+    // Log state from SpokePool
+    const originChainId = await spokePool.chainId();
```

Comment:
I'll wrap these in a multicall to speed up the queries.

Reply:
Done: https://github.com/across-protocol/contracts/pull/582/commits/d52f40f0c1df5fa0ebd3a676a9e8bcc8b2046537